File: swift day3/代码/Test/Test/ViewController.m
//
//  ViewController.m
//  Test
//
//  Created by 1 on 16/6/17.
//  Copyright © 2016年 小码哥. All rights reserved.
//

#import "ViewController.h"
#import "Student.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    /*
     __weak和__unsafe_unretained的相同点和区别
        相同点: 都不会对原有的对象进行一次retain操作
        不同点:
            1.__weak修饰的引用,当对象销毁时,会自动将指针指向nil
            2.__unsafe_unretained修饰的引用,当对象销毁时,会依然指向之前的内存地址. 注意:很容易访问僵尸对象
     */
    
    Student *stu = [[Student alloc] init];
    stu.name = @"why";
    
    __unsafe_unretained Student *stu1 = stu;
    
    stu = nil;
    
    NSLog(@"%@", stu1.name);
}

@end

File: swift day3- 语法高级/代码/13-访问权限/13-访问权限/ViewController.swift
//
//  ViewController.swift
//  13-访问权限
//
//  Created by 1 on 16/6/17.
//  Copyright © 2016年 小码哥. All rights reserved.
//

import UIKit

/*
 internal : 内部的. 默认情况下所有的访问权限都是internal
    所有的在本模块中的文件里面都可以进行访问
 private : 私有的. 
    在当前的源文件中可以访问,但是其他源文件中不能访问
 public : 可以跨模块(框架)进行访问
*/

private class ViewController: UIViewController {
    
    private var name : String = ""
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
}

class Person {
    func test() {
        ViewController().name = "why"
    }
}